cypress/integration/updateUser.spec.js

```diff
@@ -0,0 +1,731 @@
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('EmailNotValid')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, ProfileMustBeObject', function () {
+    cy.window().then((window) => {
+      const { userbase } = window
+      window._userbaseEndpoint = info.endpoint
+      userbase.init({ appId: info.appId })
+
+      return userbase.signUp(randomInfo).then((user) => {
+        cy.log(user)
+        console.log(user)
+
+        const updateInfo = {
+          profile: 'not an object',
+        }
+
+        return userbase.updateUser(updateInfo).then(() => {
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('ProfileMustBeObject')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, ProfileCannotBeEmpty', function () {
+    cy.window().then((window) => {
+      const { userbase } = window
+      window._userbaseEndpoint = info.endpoint
+      userbase.init({ appId: info.appId })
+
+      return userbase.signUp(randomInfo).then((user) => {
+        cy.log(user)
+        console.log(user)
+
+        const updateInfo = {
+          profile: {},
+        }
+
+        return userbase.updateUser(updateInfo).then(() => {
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('ProfileCannotBeEmpty')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, ProfileHasTooManyKeys', function () {
+    cy.window().then((window) => {
+      const { userbase } = window
+      window._userbaseEndpoint = info.endpoint
+      userbase.init({ appId: info.appId })
+
+      return userbase.signUp(randomInfo).then((user) => {
+        cy.log(user)
+        console.log(user)
+
+        const profileWithTooManyKeys = {}
+        for (let i = 0; i <= 1001; i++) profileWithTooManyKeys[i] = 'a'
+
+        const updateInfo = {
+          profile: profileWithTooManyKeys,
+        }
+
+        return userbase.updateUser(updateInfo).then(() => {
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('ProfileHasTooManyKeys')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, ProfileKeyMustBeString', function () {
```

Comment:
Removed the `ProfileKeyMustBeString` error from userbase-js in #141 . I'm gonna resolve this comment since  the tests I suggested in the comment are really just testing to see if JS works as expected, not if Userbase works as expected. No need for them.